embedding-stores/pgvector/deployment/src/main/java/io/quarkiverse/langchain4j/pgvector/deployment/PgVectorEmbeddingStoreProcessor.java

```diff
@@ -68,6 +72,15 @@ public void createBean(
                 .addInjectionPoint(ClassType.create(DotName.createSimple(AgroalDataSource.class)), datasourceQualifier)
                 .done());
 
+        beanProducer.produce(SyntheticBeanBuildItem
+                .configure(PG_VECTOR_AGROAL_POOL_INTERCEPTOR)
+                .types(ClassType.create(AGROAL_POOL_INTERCEPTOR))
+                .setRuntimeInit()
+                .unremovable()
+                .supplier(recorder.pgVectorAgroalPoolInterceptor())
+                .qualifiers(datasourceQualifier)
+                .done());
```

Comment:
The default scope is application-scoped?  Just making sure we don't unnecessarily create a lot of instances of the interceptor for no reason. I'm trying to find out what the default scope is, but I can't find it.

Reply:
That's a very good question. I think it is, but let's add it just to be on the safe side.

Reply:
Actually, I think `@Dependent` is the default...